common/models/role.js

```diff
@@ -29,42 +29,55 @@ module.exports = function(Role) {
   // Set up the connection to users/applications/roles once the model
   Role.once('dataSourceAttached', function() {
     var roleMappingModel = this.RoleMapping || loopback.getModelByType(RoleMapping);
+
+    /**
+     * Fetch the ids of all users assigned to this role
+     * @param  {Function} callback
+     */
     Role.prototype.users = function(callback) {
-      roleMappingModel.find({where: {roleId: this.id,
-        principalType: RoleMapping.USER}}, function(err, mappings) {
+      roleMappingModel.find({
+        where: {roleId: this.id, principalType: RoleMapping.USER}
+      }, function(err, mappings) {
         if (err) {
-          if (callback) callback(err);
-          return;
+          return callback(err);
         }
-        return mappings.map(function(m) {
+        callback(null, mappings.map(function(m) {
           return m.principalId;
```

Comment:
Now that I think about it more, another concern is that the user is being forced to load every single document associated with the Role. Imagine there were 10,000 User documents associated with a role, a reasonable scenario. This function would be unusable unless there was a way to implement pagination. Maybe we should accept a query param instead of fields so that the user could use limit, skip, and order.